chips/nrf52/src/qdec.rs

```diff
@@ -0,0 +1,278 @@
+        (0x204 => _reserved3),
+        /// Enable interrupt
+        (0x304 => intenset: ReadWrite<u32, Inte::Register>),
+        /// Disable Interrupt
+        (0x308 => intenclr: ReadWrite<u32, Inte::Register>),
+        (0x30C => _reserved4),
+        /// Enable the quad decoder
+        (0x500 => enable: ReadWrite<u32, Task::Register>),
+        /// Set the LED output pin polarity
+        (0x504 => ledpol: WriteOnly<u32, LedPol::Register>),
+        /// Sampling-rate register
+        (0x508 => sample_per: WriteOnly<u32, SampPer::Register>),
+        /// Sample register (receives all samples)
+        (0x50C => sample: WriteOnly<u32, Sample::Register>),
+        /// Reportper
+        (0x510 => report_per: ReadOnly<u32, ReportPer::Register>),
+        /// Accumulating motion-sample values register
+        (0x514 => acc: ReadOnly<u32, Acc::Register>),
+        (0x518 => acc_read: ReadOnly<u32, Acc::Register>),
+        (0x51C => reserved6),
+        (0x520 => psel_a: ReadWrite<u32, PinSelect::Register>),
+        (0x524 => psel_b: ReadWrite<u32, PinSelect::Register>),
+        (0x528 => reserved5),
+        (0x544 => accdbl: ReadOnly<u32, AccDbl::Register>),
+        (0x548 => accdbl_read: ReadOnly<u32, AccDbl::Register>),
+        (0x54C => @END),
+    }
+}
+
+register_bitfields![u32,
+    Task [
+        ENABLE 0
+    ],
+    Shorts [
+        /// Write '1' to Enable shortcut on EVENTS_COMPARE\[0\] event
+        REPORTRDY_READCLRACC 0,
+        /// Write '1' to Enable shortcut on EVENTS_COMPARE\[1\] event
+        SAMPLERDY_STOP 1,
+        /// Write '1' to Enable shortcut on EVENTS_COMPARE\[2\] event
+        REPORTRDY_RDCLRACC 2,
+        /// Write '1' to Enable shortcut on EVENTS_COMPARE\[3\] event
+        REPORTRDY_STOP 3,
+        /// Write '1' to Enable shortcut on EVENTS_COMPARE\[4\] event
+        DBLRDY_RDCLRDBL 4,
+        /// Write '1' to Enable shortcut on EVENTS_COMPARE\[5\] event
+        DBLRDY_STOP 5,
+        /// Write '1' to Enable shortcut on EVENTS_COMPARE\[6\] event
+        SAMPLERDY_READCLRACC 6
+    ],
+    Event [
+        READY 0
+    ],
+    PinSelect [
+        Pin OFFSET(0) NUMBITS(5),
+        Port OFFSET(5) NUMBITS(1),
+        Connect OFFSET(31) NUMBITS(1)
+    ],
+    Inte [
+        /// Write '1' to Enable interrupt on EVENTS_COMPARE\[0\] event
+        SAMPLERDY 0,
+        /// Write '1' to Enable interrupt on EVENTS_COMPARE\[1\] event
+        REPORTRDY 1,
+        /// Write '1' to Enable interrupt on EVENTS_COMPARE\[2\] event
+        ACCOF 2,
+        /// Write '1' to Enable interrupt on EVENTS_COMPARE\[3\] event
+        DBLRDY 3,
+        /// Write '1' to Enable interrupt on EVENTS_COMPARE\[4\] event
+        STOPPED 4
+    ],
+    LedPol [
+        LedPol OFFSET(0) NUMBITS(1) [
+            ActiveLow = 0,
+            ActiveHigh = 1
+        ]
+    ],
+    Sample [
+        SAMPLE 1
+    ],
+    SampPer [
+        SAMPLEPER OFFSET(0) NUMBITS(4) [
```

Comment:
Several of these are missing doc comments, which are useful for the auto-generated docs.

Reply:
Thanks for pointing this out! I have now added comments to all sections of the register bitfields.